src/ChatServer/ChatServer.cs

```diff
@@ -316,15 +318,20 @@ private async void ClientCleanupInactiveClients(object? sender, ElapsedEventArgs
         try
         {
             var bottomDateTimeMargin = DateTime.Now.Subtract(this.Settings.ClientTimeout);
-            for (int i = this._connectedClients.Count - 1; i >= 0; i--)
```

Comment:
Please revert this loop as well.
The parallel for will cause issues, because client.LogOffAsync mutates the list. The for loop is running backwards for this reason as well :)